StRoot/StFwdTrackMaker/StFwdTrackMaker.cxx

```diff
@@ -737,22 +742,113 @@ void StFwdTrackMaker::loadFttHitsFromGEANT( FwdDataSource::McTrackMap_t &mcTrack
     }
 } // loadFttHits
 
-void StFwdTrackMaker::loadFstHits( FwdDataSource::McTrackMap_t &mcTrackMap, FwdDataSource::HitMap_t &hitMap, int count ){
+/**
+ * @brief Loads FST hits from various sources into the hitmap and McTrackMap (if availabale)
+ * 
+ * Order of precedence:
+ * MuDst StMuFstCollection (Data)
+ * StEvent StFstHitCollection (Data or slowsim)
+ * StEvent StRndHitCollection (fast sim)
+ * GEANT St_g2t_fts_hit (starsim only) - note if rasterizer is active this takes priority over FastSim
+ * 
+ * @param mcTrackMap : MC track map if running sim
+ * @param hitMap : FST hitmap to populate
+ * @param count  : number of hits loaded
+ */
+int StFwdTrackMaker::loadFstHits( FwdDataSource::McTrackMap_t &mcTrackMap, FwdDataSource::HitMap_t &hitMap ){
+    
+    int count = loadFstHitsFromMuDst(mcTrackMap, hitMap); 
+    if ( count > 0 ) return count; // only load from one source at a time
+    
+    count += loadFstHitsFromStEvent(mcTrackMap, hitMap); 
+    if ( count > 0 ) return count; // only load from one source at a time
+    
+    bool siRasterizer = mFwdConfig.get<bool>( "SiRasterizer:active", false );
+
+    if ( !siRasterizer ) count += loadFstHitsFromStEventFastSim( mcTrackMap, hitMap );
+    if ( count > 0 ) return count; // only load from one source at a time
+
+    return loadFstHitsFromGEANT( mcTrackMap, hitMap );    
+} // loadFstHits
+
+int StFwdTrackMaker::loadFstHitsFromMuDst( FwdDataSource::McTrackMap_t &mcTrackMap, FwdDataSource::HitMap_t &hitMap){
+    int count = 0;
+    StMuDstMaker *mMuDstMaker = (StMuDstMaker *)GetMaker("MuDst");
+    if(!mMuDstMaker) {
+        LOG_WARN << " No MuDstMaker ... bye-bye" << endm;
+        return 0;
+    }
+    StMuDst *mMuDst = mMuDstMaker->muDst();
+    if(!mMuDst) {
+        LOG_WARN << " No MuDst ... bye-bye" << endm;
+        return 0;
+    }
+    
+    StMuFstCollection * fst = mMuDst->muFstCollection();
+    if (!fst) {
+        LOG_WARN << "No StMuFstCollection ... bye-bye" << endm;
+        return 0;
+    }
+
+    LOG_INFO << "Loading " << fst->numberOfHits() << " StMuFstHits" << endm;
+    TMatrixDSym hitCov3(3);
+    for ( unsigned int index = 0; index < fst->numberOfHits(); index++){
+        StMuFstHit * muFstHit = fst->getHit( index );
+
+        float vR = muFstHit->localPosition(0);
+        float vPhi = muFstHit->localPosition(1);
+        float vZ = muFstHit->localPosition(2);
+
+        const float dz0 = fabs( vZ - 151.75 );
+        const float dz1 = fabs( vZ - 165.248 );
+        const float dz2 = fabs( vZ - 178.781 );
+
+        int d = 0 * ( dz0 < 1.0 ) + 1 * ( dz1 < 1.0 ) + 2 * ( dz2 < 1.0 );
```

Comment:
Magic numbers... It took me a while to understand the meaning of 1.0 which is probably a pretty arbitrary value.